### ISSUE TITLE
elements that contain @ or $ are unparsable

### ISSUE THREAD

**Issue by ddrscott** [joshbuddy/jsonpath#18]
Unfortunately the data I'm receiving contains "@" symbols and the StringScanner utilization doesn't have a way of escaping special characters.

``` ruby
data = 
{"colors"=>
  [{"@r"=>255, "@g"=>0, "@b"=>0},
   {"@r"=>0, "@g"=>255, "@b"=>0},
   {"@r"=>0, "@g"=>0, "@b"=>255}]}

JsonPath.on(data, "$..\\@r")
# => []
# should be => [255, 0, 0] 
```

**Comment by pezra** [comment]
@Skarlso, this is still an issue that i would like fixed. I closed #25 only because it was cluttering my pull request list and it didn't look like it would ever get merged. I'd be happy to make any improvements needed to get that merged.

**Comment by Skarlso** [comment]
@pezra Could you please re-test with 0.7.0?
